src/__tests__/lib/containers/AccessApprovalsTable.test.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react'
+import { server } from '../../../mocks/msw/server'
+import { render, screen } from '@testing-library/react'
+import { AccessApprovalSearchResult } from '../../../lib/utils/synapseTypes'
+import {
+  AccessApprovalsTable,
+  AccessApprovalsTableProps,
+} from '../../../lib/containers/AccessApprovalsTable'
+import { createWrapper } from '../../../lib/testutils/TestingLibraryUtils'
+import { MOCK_USER_ID } from '../../../mocks/user/mock_user_profile'
+import {
+  mockApprovalSearchResponse,
+  mockAccessApprovalSearchResult2,
+} from '../../../mocks/MockAccessApprovals'
+
+import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'
+import { useSearchAccessApprovalsInfinite } from '../../../lib/utils/hooks/SynapseAPI/dataaccess/useAccessApprovals'
+import userEvent from '@testing-library/user-event'
+
+jest.mock(
+  '../../../lib/utils/hooks/SynapseAPI/dataaccess/useAccessApprovals',
+  () => {
+    return {
+      useSearchAccessApprovalsInfinite: jest.fn(),
+    }
+  },
+)
+
+const mockFetchNextPage = jest.fn()
+const mockSearchAccessApprovalsInfinite =
+  useSearchAccessApprovalsInfinite as jest.Mock
+
+const page2: AccessApprovalSearchResult = mockAccessApprovalSearchResult2
+
+function renderComponent(props: AccessApprovalsTableProps) {
+  render(<AccessApprovalsTable {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+describe('AccessApprovalsTable tests', () => {
+  beforeAll(() => {
+    server.listen()
+    mockSearchAccessApprovalsInfinite.mockReturnValue({
+      data: {
+        pages: [
+          {
+            results: mockApprovalSearchResponse.results,
+            nextTokenPage: mockApprovalSearchResponse.nextPageToken,
+          },
+          {
+            results: page2,
+            nextPageToken: null,
+          },
+        ],
+        pageParams: [],
+      },
+      fetchNextPage: mockFetchNextPage,
+      hasNextPage: true,
+      isLoading: false,
+      isSuccess: true,
+    })
```

Comment:
Just a note for the future, I prefer if we mock the service with MSW rather than mocking react-query's behavior; the test becomes more brittle if we modify the component to use other react-query features later.

This is fine though, so no need to change this for this PR.